### ISSUE TITLE
Min. codebraid version check failing

### ISSUE THREAD

**Issue by lcnbr** [gpoore/codebraid-preview-vscode#5]
I'm using the newest version of this extension as well as the newest version of codebraid on a windows machine through WSL. Since updating to the latest version of the preview, the codebraid version check keeps failing, and thus it is not applied. 

A notification appears saying:
>Codebraid is not installed or is out of date (need v0.7.0+). Install or upgrade (https://pypi.org/project/codebraid/), then reload the extension. Code execution and loading output from cache is disabled.

However running `codebraid --version` in the terminal yields `Codebraid 0.7.0` as required. It seems the `check_codebraid.ts` has not worked.

`

**Comment by gpoore** [comment]
It's working for me on multiple Windows machines without using WSL. I suspect there is some issue related to WSL, either in the extension or your configuration. Do you have both VS Code and Codebraid installed in WSL, or just one of them, with the other in Windows? When you run `codebraid --version`, are you using the WSL terminal or the Windows terminal?

**Comment by lcnbr** [comment]
I'm running VS Code in WSL and codebraid is installed in WSL (along with pandoc). I ran `codebraid --version` in the WSL terminal.

 I will try with an all windows setup this evening and get back to you!

**Comment by lcnbr** [comment]
Yes so it definately a WSL based issue as there are no issues with a full windows version. Maybe the way the version check is calling the shell?

**Comment by gpoore** [comment]
How do you have Python configured in WSL? Are you using conda or some other sort of environment?

At least for conda environments, it looks like subprocesses launched in VS Code extensions don't inherit the environment, so that can prevent executables from being found. I'm working on a new release that will fix the conda case. If you're using some other sort of environment, I can look into that as well.

**Comment by lcnbr** [comment]
Yes I am using conda! That must be it then. Thanks for this plugin btw it is a really well thought out extension!

**Comment by gpoore** [comment]
I've just released a new version that should fix this. If you set a Python interpreter in VS Code, then its Python installation is now checked for a `codebraid` executable. If a `codebraid` executable is found and is part of a conda environment, then `conda run` is used to run `codebraid` so that the environment is activated.

The new version requires Codebraid v0.8.0, which I just released.

**Comment by mwcraig** [comment]
I'm still having this issue (on a Mac) with `codebraid 0.8` and `codebraid-preview 0.5.1`. I can get `codebraid-preview` to run properly if I open VS Code from the command line in my environment that has `codebraid`. When I do that, I get the error message below, which seems to indicate the check for `codebraid` in the environment is still failing.

The selected interpreter is the one with `codebraid  0.8` installed.

```
The Python interpreter selected in VS Code does not have codebraid installed. Falling back to codebraid on PATH.
```

**Comment by gpoore** [comment]
@mwcraig If you can give me a little information about the layout inside the environment, that may help track down the issue.
* Where is the `codebraid` executable in the environment? Is it `<path_to_env>/bin/codebraid` or `<path_to_env>/Scripts/codebraid`, or something else?
* Is there a `conda-meta` directory in the main environment directory? That is, does `<path_to_env>/conda-meta` exist?

One more thing that might help. If you open a terminal without the environment activated, in a directory other than the environment, and run the command below (properly adjusted), does it work?
```
conda run --prefix <absolute_path_to_env> --no-capture-output <absolute_path_to_env_codebraid_executable> --version
```

**Comment by mwcraig** [comment]
> @mwcraig If you can give me a little information about the layout inside the environment, that may help track down the issue.
> 
> * Where is the `codebraid` executable in the environment? Is it `<path_to_env>/bin/codebraid` or `<path_to_env>/Scripts/codebraid`, or something else?

It is at `/Users/mattcraig/mambaforge/envs/braid/bin/codebraid` (`mambaforge` has the best M1 mac support at the moment)

> * Is there a `conda-meta` directory in the main environment directory? That is, does `<path_to_env>/conda-meta` exist?
> 

Yes

> One more thing that might help. If you open a terminal without the environment activated, in a directory other than the environment, and run the command below (properly adjusted), does it work?
> 
> ```
> conda run --prefix <absolute_path_to_env> --no-capture-output <absolute_path_to_env_codebraid_executable> --version
> ```

I get this (base env, executed in home directory):

```bash
 $ conda run --prefix  /Users/mattcraig/mambaforge/envs/braid --no-capture-output /Users/mattcraig/mambaforge/envs/braid/bin/codebraid --version
Codebraid 0.8.0
```

**Comment by gpoore** [comment]
@mwcraig All of that is equivalent to what I've seen under Windows and Linux, so there don't seem to be any issues due to conda differences between operating systems.

There is a possibility that this is related to issues in the VS Code Python extension, which is responsible for tracking the current interpreter/environment (for example, https://github.com/microsoft/vscode-python/issues/15467). To check that, there are a couple of things you might try.
* Open the markdown file you want to try. While it is open, before starting Codebraid Preview, open the `Python: Select Interpreter` command from the Command Palette (`Ctrl+Shift+P`), click the refresh button (top right, circling arrow), wait for refresh (should be a progress indicator at bottom of dropdown), and then select the environment (or enter it if not found). (More details if needed: https://code.visualstudio.com/docs/python/environments.) This should cause the Python extension to load the interpreter/environment list completely, which apparently may not happen in some cases. Once the interpreter list is updated and you have selected the environment, then start Codebraid Preview and see what happens. Even if you get a warning message, go ahead and click the "Run Codebraid" button to see what happens.
* Another test would be to make sure that the interpreter is set for the markdown file as above. Then close and restart VS Code. Open a Python file (any `*.py`), which should start the Python extension. Wait a minute or two to make sure it is completely loaded. Then open the markdown file and see what happens with Codebraid Preview.

**Comment by mwcraig** [comment]
> Open the markdown file you want to try. While it is open, before starting Codebraid Preview, open the Python: Select Interpreter command from the Command Palette (Ctrl+Shift+P), click the refresh button (top right, circling arrow), wait for refresh (should be a progress indicator at bottom of dropdown), and then select the environment (or enter it if not found). (More details if needed: https://code.visualstudio.com/docs/python/environments.) This should cause the Python extension to load the interpreter/environment list completely, which apparently may not happen in some cases. Once the interpreter list is updated and you have selected the environment, then start Codebraid Preview and see what happens. Even if you get a warning message, go ahead and click the "Run Codebraid" button to see what happens.

Tried this (both starting the Finder and from the command line) and still no luck, same message as before. There wasn't much delay (if any) when I hit refresh, but the machine is fairly fast and the number of environments is small.

> Another test would be to make sure that the interpreter is set for the markdown file as above. Then close and restart VS Code. Open a Python file (any *.py), which should start the Python extension. Wait a minute or two to make sure it is completely loaded. Then open the markdown file and see what happens with Codebraid Preview.

Still no luck (same error message about codebraid not on found).

I can run the python file and get this in VS Code's terminal:

```
conda activate braid
/Users/mattcraig/mambaforge/envs/braid/bin/python /Users/mattcraig/Downloads/hello.py

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
(base) matt-studio:~ mattcraig$ conda activate braid
(braid) matt-studio:~ mattcraig$ /Users/mattcraig/mambaforge/envs/braid/bin/python /Users/mattcraig/Downloads/hello.py
Hell world!
(braid) matt-studio:~ mattcraig$ 

```

**Comment by gpoore** [comment]
I've just released a new version that should finally fix this. I found an array indexing bug, and also added code to work around potential delays in the Python extension locating environments. The new version requires Codebraid v0.9.0, which I also just released.

If there are any future issues related to Python environments, I've added some basic logging. This is available by opening the Problems panel (View, Problems), going to the Output tab, and then selecting Codebraid Preview from the dropdown.
